### ISSUE TITLE
Possible Overfitting in Model zeus-olimpic-1.0-2024-02-12.

### ISSUE THREAD

**Issue by bdim404** [Jirka-Mayer/ICDAR2024#2]
Hi

I tested your recognition model **zeus-olimpic-1.0-2024-02-12**, and I observed a potential overfitting issue.  

- When I tested the model with a sample from the dataset, the prediction accuracy was acceptable.  
- However, when I tested it with samples outside the dataset, the model produced significant errors.  

This raises concerns about the model's generalization ability. Could you confirm whether overfitting might be affecting its performance?  

I can provide several test samples I used—these are excerpts from piano sheet music adapted from Chinese traditional folk music. 

<img width="1283" alt="Image" src="https://github.com/user-attachments/assets/dd499456-4946-4b20-be89-742afbc73323" />

<img width="768" alt="Image" src="https://github.com/user-attachments/assets/8f444f63-dca4-472a-87a3-b604175daa86" />

**Comment by Jirka-Mayer** [comment]
Hello,

the model definitely struggles with generalization to unseen domains. We did use image augmentation techniques to help it generalize in the visual-style domain, but these help only so much. It still might be brittle and break unexpectedly. Partly because in the training data, all the symbols look the same and have the same spacing - since it all comes out of MuseScore.

Looking at the sample images you've provided - they seem nice, black/white, typeset. This is what the model was trained for. So I'm guessing the issue might be with the alignment of the grandstaff inside the image itself. In the training data, the grandstaff takes up exactly 50% of the image vertically and is centered in the middle. I'd aim for the same placement when doing inference (so for example, the first image you've provided seems to be scaled up too much - it takes up about 80% of the image).

But I'm just guessing. It might be by the shape of notes and their spacing - that was NOT varied during training using image augmentation.

I tried running the two images through the model myself to verify. It indeed performs poorly. The first image is much worse than the second - my guess would be the scaling of the grandstaff within the image. The second one at least resembles the traget, although it's still a mess.

These are the two LMX strings and their corresponding MusicXML outputs:

![Image](https://github.com/user-attachments/assets/f8d9646d-8ec0-4f60-a7a8-6eace5c84c22)

```
measure key:fifths:0 time beats:2 beat-type:4 clef:G2 staff:1 clef:F4 staff:2 print-object:no rest voice:1 rest:measure staff:1 backup half A5 voice:5 eighth stem:down staff:1 beam:begin slur:start C6 eighth beam:end clef:F4 staff:2 C3 eighth dot staff:2 A3 16th stem:up beam:begin beam:begin C4 16th stem:down staff:1 beam:end beam:end measure print-object:no rest voice:1 rest:measure staff:1 backup half A5 voice:5 eighth stem:down staff:1 beam:begin C5 eighth beam:end clef:F4 staff:2 E3 16th staff:2 beam:begin beam:begin A3 16th beam:end beam:end C4 16th dot beam:begin beam:begin E4 32nd beam:end beam:end beam:backward-hook measure print-object:no rest voice:1 quarter dot staff:1 E5 16th dot stem:down beam:begin beam:begin G5 32nd beam:end beam:end beam:backward-hook backup half A5 voice:5 eighth stem:down staff:1 beam:begin C5 eighth beam:end print-object:no rest eighth staff:2 A3 eighth beam:begin measure B5 voice:1 16th dot stem:down staff:1 beam:begin beam:begin C5 32nd beam:end beam:end beam:backward-hook D5 eighth dot beam:begin C5 16th beam:end beam:backward-hook F5 16th dot beam:begin beam:begin B5 32nd beam:end beam:end beam:backward-hook backup half C4 voice:5 eighth stem:down staff:2 beam:begin E4 eighth beam:end A3 eighth beam:begin G3 eighth beam:end measure B5 voice:1 eighth dot stem:down staff:1 beam:begin D6 16th beam:forward-hook B5 eighth beam:end C6 16th beam:begin beam:begin B5 16th D6 16th G6 16th beam:end beam:end backup half E4 voice:5 eighth stem:down staff:2 chord G4 eighth rest eighth G3 eighth chord B3 eighth rest eighth measure B5 voice:1 16th stem:down staff:1 beam:begin beam:begin D6 16th G5 16th E5 16th beam:end beam:end C6 16th beam:begin beam:begin B5 16th G5 16th E5 16th beam:end beam:end backup half C4 voice:5 eighth stem:down staff:2 rest eighth G3 eighth chord B3 eighth rest eighth
```

![Image](https://github.com/user-attachments/assets/98889e66-1837-470f-b104-c537b2185497)

```
measure key:fifths:0 time beats:4 beat-type:4 clef:G2 staff:1 clef:F4 staff:2 print-object:no rest voice:1 half staff:1 print-object:no rest quarter backup whole rest voice:5 rest:measure staff:2 measure clef:F4 staff:2 F3 voice:1 16th stem:up staff:1 beam:begin beam:begin A3 16th B3 16th E4 16th beam:end beam:end A4 16th beam:begin beam:begin C4 16th A3 16th B3 16th beam:end beam:end C4 16th stem:down beam:begin beam:begin F4 16th A4 16th C5 16th beam:end beam:end backup whole print-object:no rest voice:5 rest:measure staff:2 measure B3 voice:1 16th stem:up staff:1 beam:begin beam:begin C4 16th E4 16th C5 16th beam:end beam:end D5 eighth stem:down beam:begin C5 eighth F4 16th beam:begin G4 16th beam:end beam:end C5 16th beam:begin beam:begin F5 16th G5 16th D5 16th beam:end beam:end backup whole print-object:no rest voice:5 rest:measure staff:2
```

**Comment by bdim404** [comment]
Hello,

Thank you for your detailed response! I’m currently studying your paper and learning more about the model’s training process. I’m particularly interested in whether you plan to further improve the model and training methodology in the future.

I’d like to explore ways to enhance the model myself, but if you already have plans for improvements, I’d love to know if there’s an opportunity to contribute or participate in your work. Let me know what you think!

**Comment by bdim404** [comment]
Hello,  

Thank you for your detailed response! I appreciate the insights into the model's generalization challenges and the potential impact of grandstaff alignment and note spacing on recognition performance.  

I am currently studying your paper and trying to understand the underlying architecture and training approach. I was wondering if you have any plans to further improve the model structure or training methodology. Additionally, do you intend to expand and enrich the dataset to enhance generalization?  

I am very interested in building upon this work and exploring possible improvements. If you have plans for further development, I would love to contribute and participate as a research assistant volunteer. I am passionate about this field and eager to learn more through hands-on collaboration.  

Looking forward to your thoughts :)
